sample-app/src/components/VisualAutocomplete/VisualSearchBar.tsx

```diff
@@ -50,10 +70,28 @@ export default function VisualSearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input) ?? '';
   const isLoading = useAnswersState(state => state.searchStatus.isLoading) ?? false;
-  const [executeQuery, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
+  const [executeQueryWithNearMeHandling, autocompletePromiseRef] = useSearchWithNearMeHandling(answersActions)
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(async () => {
     return answersActions.executeUniversalAutocomplete();
   });
+  const [
```

Comment:
it feels a little weird that we return an instance of the recent-searches library and not just the recent searches data object itself? If we imagine that we're not using any particular library for this, it feels a little arbitrary why setRecentSearches() and getRecentSearches() are grouped in an object, but we also provide 2 additional methods after that. I would probably shoot for something as close to a useState hook as possible.
```ts
const [ recentSearches, setRecentSearches, clearRecentSearches ] = useRecentSearches(limit);
```
or maybe even this, which would let us also factor out the effect too
```ts
const [ recentSearches, setRecentSearches ] = useRecentSearches(limit, disableRecentSearches);
```
@tmeyer2115 do you think it would be okay to pass in a disableRecentSearches boolean? It's a little weird, but I also think having a createNewRecentSearchesStorage method is also a little weird?

Reply:
I'm not crazy about the current method signature. I think the two `*Storage` methods are leaking implementation details. They also don't make the method feel particularly "hook" like as Oshi said. I'm also not crazy about adding a boolean to `useRecentSearches`. 

What about making a wrapper class over RecentSearches that has the following methods:
- get(...)
- set(...)
- clear()

The first two would just be pass-throughs. The second would clear out the corresponding bit of local storage. The VA Component could invoke `clear()` whenever `hideRecentSearches` is true. I think having a wrapper also gives us the benefit of not leaking implementation details of `RecentSearch` into other classes. We could make `RecentSearches` a singleton if necessary. I'm not saying this is the best way, but I'd live to avoid adding the boolean to `useRecentSearches` and exposing implementation details about local storage. Happy to discuss further. @oshi97 @yen-tt

Reply:
That interface sounds good to me.